### PR TITLE
Squashed commit of the following:

### PR DESCRIPTION
**Pull request by MarkMc1089**
commit b02e38b0a47f932a747e59bac09ea49c094036a5
Author: Mark McPherson <mark.mcpherson@nhsbsa.nhs.uk>
Date:   Fri Nov 3 09:30:20 2023 +0000

    Remove gitleaksignore lines + use ** power syntax

commit 48275e0febf5ac74f53c19b2737f9fde676df45a
Author: Mark McPherson <mark.mcpherson@nhsbsa.nhs.uk>
Date:   Thu Nov 2 16:56:07 2023 +0000

    Ignore false +ve gitleaks

commit 6ba1cdcfd2ab45ed2ba55aa4d7336049449bb4e9
Author: Mark McPherson <mark.mcpherson@nhsbsa.nhs.uk>
Date:   Thu Nov 2 16:45:28 2023 +0000

    Add totals for items and costs

commit 134b51e2371e3ad510be504d5cda4976751f0991
Author: Mark McPherson <mark.mcpherson@nhsbsa.nhs.uk>
Date:   Wed Nov 1 16:25:29 2023 +0000

    WIP

### CODE REVIEW

File: .gitleaksignore [hunk 1]
```diff
@@ -1 +0,0 @@
-018bcb19a754f9a34f0866ac247e89b2de1cf839:data-raw/app/calc_missing_address_paper_rx.R:nhsnumber:31
```
Comment (MarkMc1089) [inline .gitleaksignore:1]: I had some trouble with gitleaks (all false +ve's) at first. While trying to sort, I noticed this no longer needed ignore - the file referred to is no longer in the repo.

File: DESCRIPTION [hunk 2]
```diff
@@ -46,6 +46,7 @@ Imports:
     reactable,
     rlang,
     rsvg,
+    scales,
```
Comment (MarkMc1089) [inline DESCRIPTION:49]: Provides easy way to format mixes of short and long numbers.

File: R/mod_07_ch_flag_drug.R [hunk 3]
```diff
@@ -148,11 +149,77 @@ mod_07_ch_flag_drug_server <- function(id, export_data) {
...
+                    return (this.value / 10**6) + 'M';
+                }
+                else if(this.value >= 10**3) {
+                    return (this.value / 10**3) + 'K';
+                }
+                else {
+                    return this.value;
+                }
+              }
+            ")
```
Comment (MarkMc1089) [inline R/mod_07_ch_flag_drug.R:169]: Handle ranges from small to large, to avoid axis labels like `1200000000`!

File: R/mod_07_ch_flag_drug.R [hunk 4]
```diff
@@ -148,11 +149,77 @@ mod_07_ch_flag_drug_server <- function(id, export_data) {
...
+                    point.series.name + ': ' +
+                    prefix(point.point.METRIC) + 
+                    fmt_num(point.y) + 
+                    suffix(point.point.METRIC);
+                },
+                '<b>' + this.x + '</b>'
+              );
+            }
+            "
+          )
```
Comment (MarkMc1089) [inline R/mod_07_ch_flag_drug.R:222]: Handle both formatting and prefix/suffix all in one.

File: R/mod_08_geo_ch_flag_drug.R [hunk 5]
```diff
@@ -339,7 +339,12 @@ mod_08_geo_ch_flag_drug_server <- function(id, export_data) {
       # Process original df
       df = df %>%
         dplyr::rename_at(fy, ~"VALUE") %>%
-        dplyr::mutate(VALUE_LABEL = sprintf("%.2f", janitor::round_half_up(VALUE, 2))) %>% 
+        dplyr::mutate(
+          VALUE_LABEL = scales::label_comma(
+            accuracy = .01,
+            scale_cut = scales::cut_long_scale()
+          )(janitor::round_half_up(VALUE, 2))
+        ) %>%
```
Comment (MarkMc1089) [inline R/mod_08_geo_ch_flag_drug.R:347]: Handle small and large numbers in the spline charts

File: R/mod_08_geo_ch_flag_drug.R [hunk 6]
```diff
@@ -595,11 +618,24 @@ mod_08_geo_ch_flag_drug_server <- function(id, export_data) {
...
+            headerClass = "my-header",
+            format = reactable::colFormat(digits = 2),
+            cell = function(val, row, col_name) {
+              if (col_name %in% c(names(geographies), ".selection")) return (val)
+              
+              return (
+                scales::label_comma(
+                  accuracy = .01,
+                  scale_cut = scales::cut_long_scale()
+                )(val)
```
Comment (MarkMc1089) [inline R/mod_08_geo_ch_flag_drug.R:635]: Custom format for metric value cells; `scales::label_comma` returns a function, in this case one that rounds to 2 d.p. and uses K, M, B suffixes appropriately.

File: data-raw/app/07_ch_flag_drug_df.R [hunk 7]
```diff
@@ -75,10 +74,12 @@ get_geo_bnf_prop = function(index){
       BNF_PARENT = rlang::as_string(bnf),
       BNF_CHILD := {{ bnf }},
       PROP_ITEMS = janitor::round_half_up(PROP_ITEMS, 3),
-      PROP_NIC = janitor::round_half_up(PROP_NIC, 3)
+      PROP_NIC = janitor::round_half_up(PROP_NIC, 3),
+      TOTAL_ITEMS = ITEMS,
```
Comment (MarkMc1089) [inline data-raw/app/07_ch_flag_drug_df.R:78]: I only needed to keep the already calculated totals, include them in the pivot and assign a human-friendly name to them.

File: R/mod_07_ch_flag_drug.R [hunk 8]
```diff
@@ -106,15 +106,6 @@ mod_07_ch_flag_drug_server <- function(id, export_data) {
...
 
-    # Metrics
-    p1 = "% of total annual number of prescription items"
-    p2 = "% of total annual drug cost"
-    c1 = "Mean drug cost PPM"
-
-    # Pound sign for pound metrics
-    prefix = reactive({ifelse(input$input_metric == c1, "£", "")})
-    suffix = reactive({ifelse(input$input_metric %in% c(p1,p2), "%", "")})
-
```
Comment (MarkMc1089) [inline R/mod_07_ch_flag_drug.R:117]: Moved logic into the tooltip formatter JS.

**Comment by AdnanShroufi** [comment]
All looks good, for what I can see, are some changes listed elsewhere?

**Comment by MarkMc1089** [comment]
> All looks good, for what I can see, are some changes listed elsewhere?

There are ~7 commits previous to the ones from yesterday, most right at top

**Comment by AdnanShroufi** [comment]
All files reviewed and ready to merge, only thing required is the side tab for 'Updates' (i.e. we have now added 2 metrics to mod7 and mod8 since the analysis was publicly released)

**Pull request merged**